client.go

```diff
@@ -57,6 +67,12 @@ func NewClientWithURL(token, url string) (Client, Me, error) {
 	return client, me, nil
 }
 
+// AddAblyAgent adds an extra entry to the value sent as the Ably-Agent HTTP
+// header.
+func (c *Client) AddAblyAgent(product, version string) {
```

Comment:
With the current implementation of this method, whereby the `ablyAgent` field is overwritten on each call, I would call it `SetAdditionalAblyAgent`... or, to keep this name, perhaps have an `ablyAgents` array field (more akin to [`RSC7d6`](https://sdk.ably.com/builds/ably/specification/main/features/#RSC7d6)).

Reply:
I did initially go with an array, and `strings.Join(c.ablyAgents, " ")` on each request, but the Paddy in me told me I should only construct the string once to avoid per-request allocations 🙂

Also, I'm not sure I like `SetAdditionalAblyAgent`, how about `AppendAblyAgent`?

Reply:
Renamed to `AppendAblyAgent` in 15474d4.